packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -166,9 +173,33 @@ export async function execCommand(command: string, options: SpawnOptions = {}) {
   return true;
 }
 
-export async function getNodeBinPath({ cwd }: { cwd: string }) {
-  const { stdout } = await execAsync('npm', ['bin'], { cwd });
-  return stdout.trim();
+export async function getNodeBinPath({
+  cwd,
+}: {
+  cwd: string;
+}): Promise<string | undefined> {
+  const { code, stdout, stderr } = await execAsync('npm', ['bin'], {
+    cwd,
+    prettyCommand: 'npm bin',
+    ignoreNon0Exit: true,
+  });
+
+  // in some rare cases, we saw `npm bin` exit with code 7, but still
+  // output the right bin path, so we consider both code 0 and 7 as success
+  if (code !== 0 && code !== 7) {
```

Comment:
What about if we ignored the exit code and instead simplified this: 

```js
const nodeBinPath = stdout.trim()

if (path.isAbsolute(nodeBinPath)) {
  return nodeBinPath
}

throw new NowBuildError()
```

Reply:
I like it too 👍

Reply:
Updated in https://github.com/vercel/vercel/pull/8058/commits/8d448a82080910aef52f1a2704b802ac7bf09f38 👍 Thank you for the suggestion!